### DISCUSSION TITLE
Transparent clickthrough window

### DISCUSSION THREAD

**Discussion by TheCommCraft** [openrndr/openrndr#409] [Q&A]
How can I create a transparent clickthrough window with openrndr?

**Comment by hamoid** [comment]
Hi! Never seen such a thing yet :) 

Do you have an example of such a program?

For what OS?

**Comment by TheCommCraft** [comment]
Like described in this article
https://www.codeproject.com/Articles/12365/How-to-make-a-transparent-and-click-through-window

**Comment by hamoid** [comment]
I see :) A starting point: https://stackoverflow.com/questions/10276020/lwjgl-window-with-transparent-background

**Reply by TheCommCraft** [reply]
So it's not possible just with openrndr?

**Reply by hamoid** [reply]
OPENRNDR uses LWJGL. Maybe it is possible by modifying the core libraries, but it would need some work to be done by whoever is interested in making this happen :)

**Comment by TheCommCraft** [comment]
Well I went to openrndr specifically to avoid having to deal with OpenGL
but I'll have a look

**Reply by hamoid** [reply]
It is a cool feature, but maybe rarely needed? How do you plan to use it?

**Comment by TheCommCraft** [comment]
I would like to try and add distortions to the whole screen

Abe Pazos ***@***.***> schrieb am Do., 3. Okt. 2024, 13:36:

**Comment by edwinRNDR** [comment]
Looks like GLFW supports this. It should not be too hard to hack this into OPENRNDR's `ApplicationGLFWGL3` and see what it does for you.

> GLFW_MOUSE_PASSTHROUGH specifies whether the window is transparent to mouse input, letting any mouse events pass through to whatever window is behind it. This is only supported for undecorated windows. Decorated windows with this enabled will behave differently between platforms. Possible values are GLFW_TRUE and GLFW_FALSE.
